collisions.js

```diff
@@ -0,0 +1,116 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+'use strict';
+
+var createTable = require('./lib/table.js').create;
+var ClusterManager = require('./lib/cluster.js');
+var parseCollisionCommand = require('./parser.js').parseCollisionCommand;
+var detectChecksumMethod = require('./lib/hash-method.js');
+
+function main() {
+    var command = parseCollisionCommand();
+    var clusterManager = new ClusterManager({
+        useTChannelV1: command.useTChannelV1,
+        discoveryUri: command.discoveryUri
+    });
+
+    printCollisions(command, clusterManager);
+}
+
+function printCollisions(command, clusterManager) {
+    clusterManager.fetchStats(function onStats(err) {
+        if (err) {
+            console.error('Error: ' + err.message);
+            process.exit(1);
+        }
+
+        var partitions = clusterManager.getPartitions();
+
+        var hashFunction = null;
+        var memberHashes = {};
+        var replicaHashes = {};
+
+        var memberCollisionTable = createTable(['hash', 'address', 'collision']);
+        var replicaCollisionTable = createTable(['hash', 'address', '# replica', 'collision', '# replica']);
+
+        function checkMemberCollision(address) {
+            var hash = '' + hashFunction(address);
+            if (!memberHashes[hash]) {
+                memberHashes[hash] = address;
+            } else if (memberHashes[hash] !== address) {
+                //collision!
+                memberCollisionTable.push([hash, address, memberHashes[hash]]);
+            }
+        }
+
+        function checkReplicaCollision(address, replica) {
+            var replicaName = address + replica;
+            var hash = '' + hashFunction(replicaName);
+            if (!replicaHashes[hash]) {
+                replicaHashes[hash] = {address: address, replica: replica};
+            } else if (replicaHashes[hash].address !== address) {
+                //collision!
+                replicaCollisionTable.push([hash, address, replica, replicaHashes[hash].address, replicaHashes[hash].replica]);
+            }
+        }
+
+        partitions.forEach(function eachPartition(partition) {
```

Comment:
I think this command should only be ran if the cluster is in a steady state. That means we don't need to iterate over the partition and would make the logic and output a lot easier to digest as well. Do other ringpop-admin commands deal with this in a similar manner?